src/renderer/wg_engine/tvgWgRenderer.cpp

```diff
@@ -245,10 +245,11 @@ bool WgRenderer::clear()
 
 bool WgRenderer::sync()
 {
+    if (!surface) return false;
     disposeObjects();
```

Comment:
if the canvas is destroyed in the middle of the usage, no resource leaks?

Reply:
Objects will be destoyed when renderer will be released.
But we can remove them in any case